tests/unit/confidant/authnz/authnz_test.py

```diff
@@ -209,3 +210,57 @@ def test_redirect_to_logout_if_no_auth(self):
             u_mock.is_authenticated = Mock(return_value=False)
             u_mock.redirect_to_goodbye = Mock(return_value='redirect_return')
             self.assertEqual(wrapped(), 'redirect_return')
+
+
+class HeaderAuthenticatorTest(unittest.TestCase):
+
+    def setUp(self):
+        # Save old values
+        self.username_header = app.config['HEADER_AUTH_USERNAME_HEADER']
+        self.email_header = app.config['HEADER_AUTH_EMAIL_HEADER']
+        self.first_name_header = app.config['HEADER_AUTH_FIRST_NAME_HEADER']
+        self.last_name_header = app.config['HEADER_AUTH_LAST_NAME_HEADER']
+        self.user_mod = authnz.user_mod
+
+        # Update config
+        app.config['USE_AUTH'] = True
+        app.config['USER_AUTH_MODULE'] = 'header'
+        app.config['HEADER_AUTH_USERNAME_HEADER'] = 'X-Confidant-Username'
+        app.config['HEADER_AUTH_EMAIL_HEADER'] = 'X-Confidant-Email'
+
+        # Reset the user module in use
+        authnz.user_mod = authnz.userauth.init_user_auth_class()
+
+    def tearDown(self):
+        app.config['HEADER_AUTH_USERNAME_HEADER'] = self.username_header
+        app.config['HEADER_AUTH_EMAIL_HEADER'] = self.email_header
+        app.config['HEADER_AUTH_FIRST_NAME_HEADER'] = self.first_name_header
+        app.config['HEADER_AUTH_LAST_NAME_HEADER'] = self.last_name_header
+
+        authnz.user_mod = self.user_mod
+
+    def test_will_extract_from_request(self):
```

Comment:
This is technically an integration test. Since it doesn't require dynamo or kms access we can leave it in until we get the integration tests working properly in travis.